File: Facultate/Anul 2/Sem 2/MPP/Laburi/JavaTema4/JavaTema4/Server/src/main/java/server/ObjectConcurrentServer.java
package server;

import utils.ServiceFactory;

import java.io.IOException;
import java.net.ServerSocket;
import java.net.Socket;
import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

public class ObjectConcurrentServer {
    private final int port;
    private volatile boolean running;
    private ServerSocket serverSocket;
    private final ExecutorService executor = Executors.newFixedThreadPool(20);
    private final Logger logger = LogManager.getLogger(ObjectConcurrentServer.class);

    public ObjectConcurrentServer(int port) {
        this.port = port;
    }

    public void start() throws IOException {
        serverSocket = new ServerSocket(port);
        running = true;
        logger.info("Server started on port {}", port);

        while (running) {
            Socket client = serverSocket.accept();
            executor.execute(new ClientObjectWorker(client));
        }
    }

    public void stop() throws IOException {
        running = false;
        executor.shutdownNow();
        if (serverSocket != null && !serverSocket.isClosed()) {
            serverSocket.close();
        }
        logger.info("Server stopped");
    }
}

File: Facultate/Anul 2/Sem 2/MPP/Laburi/Edi/CSharpServer/Server/server/ObjectConcurrentServer.cs
using System.Net;
using System.Net.Sockets;

namespace Server.server;

public class ObjectConcurrentServer
{
    private readonly int _port;
    private volatile bool _running;
    private TcpListener _server;

    public ObjectConcurrentServer(int port)
    {
        _port = port;
    }

    public void Start()
    {
        _server = new TcpListener(IPAddress.Any, _port);
        _server.Start();
        _running = true;

        try
        {
            while (_running)
            {
                var client = _server.AcceptSocket();

                var worker = new ClientObjectWorker(client);

                var thread = new Thread(() =>
                {
                    try
                    {
                        worker.Run();
                    }
                    catch (Exception ex)
                    {
                        Console.WriteLine("WORKER CRASH: " + ex);
                    }
                });

                thread.IsBackground = true;
                thread.Start();
            }
        }
        catch (Exception ex)
        {
            Console.WriteLine("SERVER LOOP CRASH: " + ex);
        }
    }

    public void Stop()
    {
        _running = false;
        _server.Stop();
    }
}